sdk/log/simple.go

```diff
@@ -55,7 +94,22 @@ func (s *SimpleProcessor) OnEmit(ctx context.Context, r *Record) error {
 		simpleProcRecordsPool.Put(records)
 	}()
 
-	return s.exporter.Export(ctx, *records)
+	err := s.exporter.Export(ctx, *records)
+
+	if s.selfObservabilityEnabled {
+		if err != nil {
```

Comment:
We overlooked this scenario: https://github.com/open-telemetry/opentelemetry-go/pull/7127/files#diff-a16aafee8a2d35b42b666d04285da444db534fad4a37e153ad24fd487b997e0aL46

Reply:
Good catch! I have improve tests to cover this case.